lib/kubernetes-deploy/resource_watcher.rb

```diff
@@ -12,38 +12,57 @@ def initialize(resources, logger:, deploy_started_at: Time.now.utc)
       @deploy_started_at = deploy_started_at
     end
 
-    def run(delay_sync: 3.seconds)
-      delay_sync_until = Time.now.utc
+    def run(delay_sync: 3.seconds, reminder_interval: 30.seconds)
+      delay_sync_until = last_message_logged_at = Time.now.utc
 
       while @resources.present?
         if Time.now.utc < delay_sync_until
           sleep(delay_sync_until - Time.now.utc)
         end
-        watch_time = (Time.now.utc - @deploy_started_at).round(1)
         delay_sync_until = Time.now.utc + delay_sync # don't pummel the API if the sync is fast
+
         @resources.each(&:sync)
         newly_finished_resources, @resources = @resources.partition(&:deploy_finished?)
 
-        new_success_list = []
-        newly_finished_resources.each do |resource|
-          if resource.deploy_failed?
-            @logger.error("#{resource.id} failed to deploy after #{watch_time}s")
-          elsif resource.deploy_timed_out?
-            @logger.error("#{resource.id} deployment timed out")
-          else
-            new_success_list << resource.id
-          end
+        if newly_finished_resources.present?
+          watch_time = (Time.now.utc - @deploy_started_at).round(1)
+          report_what_just_happened(newly_finished_resources, watch_time)
+          report_what_is_left(reminder: false)
+          last_message_logged_at = Time.now.utc
+        elsif due_for_reminder?(last_message_logged_at, reminder_interval)
+          report_what_is_left(reminder: true)
+          last_message_logged_at = Time.now.utc
```

Comment:
It is possible to for both things to be false (i.e. nothing happened this cycle, and we weren't due for a reminder), so that'd have to be wrapped in `if newly_finished_resources.present? || due_for_reminder?(...)`